test/dialog_complex_test.rb

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env rspec
+# encoding: utf-8
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require_relative "test_helper"
+
+require "yast2/system_service"
+require "yast2/compound_service"
+
+Yast.import "Wizard"
+Yast.import "SambaServer"
+Yast.import "SambaService"
+
+describe "SambaServerComplexInclude" do
+  class TestComplexDialog
+    include Yast::I18n
+    include Yast::UIShortcuts
+
+    def initialize
+      Yast.include self, "samba-server/complex.rb"
+    end
+  end
+
+  before do
+    allow(Yast2::SystemService).to receive(:find).with(anything).and_return(service)
+    allow(Yast2::CompoundService).to receive(:new).and_return(services)
+    allow(services).to receive(:action).and_return(action)
+  end
+
+  let(:service) { instance_double("Yast2::SystemService", save: true, is_a?: true) }
+  let(:services) { instance_double("Yast2::CompoundService", save: true) }
+  let(:action) { :start }
+
+  describe "#WriteDialog" do
+    subject(:samba) { TestComplexDialog.new }
+
+    let(:connected_users) { [1, 2] }
```

Comment:
It is not important, but reading the code I would say that `ConnectedUsers` returns a set of user names.

Reply:
You are right... I forgot to change this piece of code that I used only for _test_ the test. I am going to update it.